mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/SaveQueryGenerator.java

```diff
@@ -0,0 +1,144 @@
+    }
+    VariableElement parameter = parameters.get(0);
+    parameterName = parameter.getSimpleName().toString();
+    parameterType = parameter.asType();
+    entityDefinition = context.getEntityDefinitions().get(parameterType);
+    if (entityDefinition == null) {
+      context.getMessager().error(methodElement, "Missing definition of entity %s", parameterType);
+      throw new SkipGenerationException();
+    }
+  }
+
+  // Get the type of the given parameter, expecting a non-generic class
+  private static ClassName getType(VariableElement parameter, GenerationContext context) {
+    TypeMirror mirror = parameter.asType();
+    Element element = context.getTypeUtils().asElement(mirror);
+    if (element == null || element.getKind() != ElementKind.CLASS) {
+      context.getMessager().error(parameter, "Expected entity class, got %s", mirror);
+      throw new SkipGenerationException();
+    }
+    TypeElement typeElement = (TypeElement) element;
+    if (typeElement.getAnnotation(Table.class) == null) {
+      context
+          .getMessager()
+          .error(
+              parameter,
+              "Should be an instance of a class annotated with %s",
+              Table.class.getSimpleName());
+    }
+    return ClassName.get(typeElement);
+  }
+
+  @Override
+  public void addMembers(TypeSpec.Builder classBuilder) {
+    addPreparedStatementField(classBuilder);
+    addMethodImplementation(classBuilder);
+  }
+
+  private void addPreparedStatementField(TypeSpec.Builder classBuilder) {
+    classBuilder.addField(
+        FieldSpec.builder(
+                PreparedStatement.class,
+                preparedStatementFieldName,
+                Modifier.PRIVATE,
+                Modifier.FINAL)
+            .build());
+  }
+
+  private void addMethodImplementation(TypeSpec.Builder classBuilder) {
+    MethodSpec.Builder methodBuilder =
+        MethodSpec.methodBuilder(methodName)
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .addParameter(ClassName.get(parameterType), parameterName)
+            .addStatement(
+                "$T boundStatement = $L.bind()", BoundStatement.class, preparedStatementFieldName);
+
+    for (FieldDefinition field : entityDefinition) {
+      methodBuilder.addStatement(
+          "boundStatement.set($S, $L.$L(), $T.class)",
+          field.getColumnName(),
+          parameterName,
+          field.getGetterName(),
+          field.getType());
+    }
+    methodBuilder.addStatement("session.execute(boundStatement, $T.SYNC)", Statement.class);
+
+    classBuilder.addMethod(methodBuilder.build());
+  }
+
+  @Override
+  public void addConstructorInstructions(MethodSpec.Builder constructorBuilder) {
+    String queryName = methodName + "Query";
+    constructorBuilder.addCode(
+        "$[String $L = $T.insertInto($S)", queryName, QueryBuilderDsl.class, tableName);
+    for (FieldDefinition field : entityDefinition) {
+      constructorBuilder.addCode(
```

Comment:
looking at the generated code:
`this.save_preparedStatement = session.execute(new DefaultPrepareRequest(saveQuery), PrepareRequest.SYNC);`
I have an impression that error handling (generating error handling code) will be not a trivial task. Do you know the best pattern for errror handling with this approach?

Reply:
In that case, I think errors encountered while preparing should be rethrown to the caller of the mapper builder.
If we later have methods that call `prepare()` each time instead of caching the statement as a DAO field, then errors should be rethrown to the callers of those methods.